File: CS/04_Shapes/GetAscentAndDescentOfText.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text;
using System.Windows.Forms;
using Spire.Presentation;

namespace GetAscentAndDescentOfText
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();            
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            // Create a new Presentation object
            Presentation ppt = new Presentation();

            // Load a PowerPoint file from a specified location
            ppt.LoadFromFile(@"..\..\..\..\..\..\..\Data\GetAscentAndDescentOfText.pptx");

            // Create a StringBuilder to store text information
            StringBuilder builder = new StringBuilder();

            // Access the first slide in the presentation
            ISlide slide = ppt.Slides[0];

            // Access the first AutoShape in the slide
            IAutoShape autoshape = slide.Shapes[0] as IAutoShape;

            // Retrieve the layout lines from the TextFrame of the AutoShape
            IList<LineText> lines = autoshape.TextFrame.GetLayoutLines();

            // Iterate through each layout line
            for (int i = 0; i < lines.Count; i++)
            {
                // Get the ascent and descent properties of the current line
                float ascent = lines[i].Ascent;
                float descent = lines[i].Descent;

                // Append information about the line, ascent, and descent to the StringBuilder
                builder.AppendLine("lines" + i + "\tascent: " + ascent + "\tdescent: " + descent);
            }

            // Specify the name for the result file
            string result = "GetAscentAndDescentOfText.txt";

            // Save to the text file
            File.WriteAllText(result, builder.ToString());

            // Dispose of the Presentation object to release resources
            ppt.Dispose();

            // Launch the result file
            PresentationDocViewer(result);
        }
	
		private void PresentationDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
    }
}

File: VB/04_Shapes/GetAscentAndDescentOfText.vb
Imports System.IO
Imports System.Text
Imports Spire.Presentation

Namespace GetAscentAndDescentOfText
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
			' Create a new Presentation object
			Dim ppt As New Presentation()

			' Load a PowerPoint file from a specified location
			ppt.LoadFromFile("..\..\..\..\..\..\..\Data\GetAscentAndDescentOfText.pptx")

			' Create a StringBuilder to store text information
			Dim builder As New StringBuilder()

			' Access the first slide in the presentation
			Dim slide As ISlide = ppt.Slides(0)

			' Access the first AutoShape in the slide
			Dim autoshape As IAutoShape = TryCast(slide.Shapes(0), IAutoShape)

			' Retrieve the layout lines from the TextFrame of the AutoShape
			Dim lines As IList(Of LineText) = autoshape.TextFrame.GetLayoutLines()

			' Iterate through each layout line
			For i As Integer = 0 To lines.Count - 1
				' Get the ascent and descent properties of the current line
				Dim ascent As Single = lines(i).Ascent
				Dim descent As Single = lines(i).Descent

				' Append information about the line, ascent, and descent to the StringBuilder
				builder.AppendLine("lines" & i & vbTab & "ascent: " & ascent & vbTab & "descent: " & descent)
			Next i

			' Specify the name for the result file
			Dim result As String = "GetAscentAndDescentOfText.txt"

			' Save to the text file
			File.WriteAllText(result, builder.ToString())

			' Dispose of the Presentation object to release resources
			ppt.Dispose()

			' Launch the result file
			PresentationDocViewer(result)
		End Sub

		Private Sub PresentationDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
	End Class
End Namespace